### ISSUE TITLE
Inline anchors in section titles are not registered in catalog

### ISSUE THREAD

**Issue by slonopotamus** [asciidoctor/asciidoctor#3633]
Convert the following document using Asciidoctor.

```adoc
= Title

Link to <<foo>>.

== [[foo]]Section 1

Text.

== [[foo]]Section 2

Text.
```

Expected: Asciidoctor complains with `id assigned to anchor already in use: foo` from [catalog_inline_anchor function](https://github.com/asciidoctor/asciidoctor/blob/v2.0.10/lib/asciidoctor/parser.rb#L1136).

Actual: document converts silently.

asciidoctor-epub3 relies on catalog to resolve document xrefs, so it is unable do resolve inline anchors in section titles because they're not registered in catalog: asciidoctor/asciidoctor-epub3#331

**Comment by mojavelinux** [comment]
The underlying problem here is the placement of the anchor. Putting an anchor at the start of the title is a non-standard location. For it to be recognized by the core parser as an ID, it must be placed at the end of the section title.

```
== Section 2 [[foo]]
```

The anchor at the beginning is treated like any other inline anchor, meaning it's just a jump point. It's not associated with the section title in the model.

**Comment by mojavelinux** [comment]
The secondary problem is that the same anchor is being used twice. That's also illegal.

**Comment by slonopotamus** [comment]
WRT anchor placement - that's what users do. They wanted to have more than one anchor on the section and that's how they achieved it.

I used same anchor twice just to demonstrate that they are not handled by catalog.

**Comment by mojavelinux** [comment]
You are correct that they are not handled by the catalog because they are not designed to be referencable internally. The are only designed for referencing from the URL.

There is an open issue about being able to have multiple labels on a section (though we don't really have a proper term for it), but that's going to have to be run through the spec committee since its a core language change.

**Comment by mojavelinux** [comment]
Internally, right now, each block can only have a single ID for referencing from other nodes. The reference table is designed with this assumption in mind.

**Comment by slonopotamus** [comment]
@programandala-net tagging you here as an interested party.

**Comment by programandala-net** [comment]
> The underlying problem here is the placement of the anchor. Putting an anchor at the start of the title is a non-standard location. For it to be recognized by the core parser as an ID, it must be placed at the end of the section title.
> 
> ```
> == Section 1 [[foo]]
> ```
> 
> (note that it is offset by a space, which is required)
> 
> The anchor at the beginning is treated like any other inline anchor, meaning it's just a jump point. It's not associated with the section title in the model.

The tool that I wrote to build the documentation of my software puts anchors at the start of the title, because I needed a common anchor for the first homonym of the glossary, and a specific anchor for each entry:

```
=== a

[#entry61-src-lib-assembler-fs]
==== [[entry61]]a
(...)

[#entry61-src-lib-memory-address_register-fs]
==== a
(...)
```

It worked fine, until attribute `epub-chapter-level` was recently introduced in Asciidoctor EPUB3. In fact there's no restriction
